### PR TITLE
Add branch as an optional field to RepoRef

### PR DESCRIPTION
**Pull request by jessitron**
We often want to specify both this and the SHA
especially for editors; they must have a branch. and statuses must have a sha.

### CODE REVIEW

File: src/operations/common/GitHubRepoRef.ts [hunk 1]
```diff
@@ -15,6 +16,18 @@ export const GitHubDotComBase = "https://api.github.com";
  */
 export class GitHubRepoRef extends AbstractRepoRef {
 
+    public static for(params: { owner: string, repo: string, sha?: string, rawApiBase?: string, path?: string, branch?: string }): GitHubRepoRef {
```
Comment (johnsonr) [inline src/operations/common/GitHubRepoRef.ts]: Not keen on using a reserved word as a function name even if it is legal

File: src/operations/support/editorUtils.ts [hunk 2]
```diff
@@ -92,7 +92,7 @@ function doWithEditResult(r: EditResult<GitProject>, gitop: () => Promise<EditRe
                 }) : gitop();
             });
     }
-    logger.info("NOT committing %j as it's not dirty, edited=%s", r.target.id, r.edited);
+    logger.info("NOT committing %j in %s as it's not dirty, edited=%s", r.target.id, r.target.baseDir, r.edited);
```
Comment (johnsonr) [inline src/operations/support/editorUtils.ts]: I think we should add logging on all the branches in this function. It's important information and will be invaluable for debugging

File: src/operations/common/GitHubRepoRef.ts (hunk 1 above)
Comment (jessitron) [reply src/operations/common/GitHubRepoRef.ts]: oh good call

File: src/operations/support/editorUtils.ts (hunk 2 above)
Comment (jessitron) [reply src/operations/support/editorUtils.ts]: ok. yeah this came up for us as I was working on this so I threw it in

**Pull request merged**